6-compiler.js

```diff
@@ -0,0 +1,27 @@
+"use strict";
+import { tokenizer } from './1-tokenizer.js';
+import { parser } from './2-parser.js';
+import { transformer } from './4-transformer.js';
+import { codeGenerator } from './5-code-generator.js';
+
+const button = document.getElementById('button');
+const input = document.getElementById('input');
+const text = document.getElementById('text');
+
+button.addEventListener('click', function() {
+  let inputValue = input.value.trim();
+  
+  let startStr = inputValue.slice( 0,1 ) ;
+  let endStr = inputValue.slice( -1 ) ;
+
+  if (startStr === '(' && endStr === ')'){
```

Comment:
If you just want to check if the input starts and ends with parentheses, it might be easier just to use a Regex test e.g. `/\(.*\)/.test(inputValue)`.

Reply:
Yes it is true.
I will correct it😇